src/app/components/shared/audio-event-card/annotation-event-card.component.ts

```diff
@@ -50,9 +50,17 @@ export class AnnotationEventCardComponent implements AfterViewInit {
   private readonly spectrogram =
     viewChild<ElementRef<SpectrogramComponent>>("spectrogram");
 
-  public ngAfterViewInit(): void {
-    setTimeout(() => {
-      this.mediaControls().nativeElement.for = this.spectrogram().nativeElement;
-    }, 0);
+  public constructor() {
+    // Use effect() to link media controls to the spectrogram when it becomes
+    // available. This is necessary because the spectrogram is inside a @defer
+    // block and won't exist until the element enters the viewport.
+    effect(() => {
+      const spectrogramEl = this.spectrogram();
+      const mediaControlsEl = this.mediaControls();
```

Comment:
Do not use shorthand names, use `spectrogramElement` and `mediaControlsElement`

```suggestion
      const spectrogramElement = this.spectrogram();
      const mediaControlsElement = this.mediaControls();
```